src/wcmCommon.c

```diff
@@ -1843,27 +1852,50 @@ TEST_CASE(test_rebase_pressure)
 {
 	WacomDeviceRec priv = {0};
 	WacomDeviceRec base = {0};
+	WacomCommonRec common = {0};
 	WacomDeviceState ds = {0};
 	int pressure;
 
-	priv.minPressure = 4;
-	ds.pressure = 10;
+	priv.common = &common;
+	common.wcmMaxZ = 2000;
 
 	/* Pressure in out-of-proximity means get new preloaded pressure */
 	priv.oldState.proximity = 0;
 
+	/* pressure < pressure_threshold */
+	ds.pressure = 10;
+
 	/* make sure we don't touch priv, not really needed, the compiler should
 	 * honor the consts but... */
 	base = priv;
 
 	pressure = rebasePressure(&priv, &ds);
-	assert(pressure == ds.pressure);
+	assert(pressure > ds.pressure);
+	assert(memcmp(&priv, &base, sizeof(priv)) == 0);
+
+	/* pre-registered pressure > pressure_threshold */
+	ds.pressure = 100;
+
+	base = priv;
 
+	pressure = rebasePressure(&priv, &ds);
+	assert(pressure == ds.pressure);
 	assert(memcmp(&priv, &base, sizeof(priv)) == 0);
 
 	/* Pressure in-proximity means rebase to new minimum */
 	priv.oldState.proximity = 1;
 
+	priv.minPressure = 20;
+	ds.pressure = 10;
+
+	base = priv;
+
+	pressure = rebasePressure(&priv, &ds);
+	assert(pressure == priv.minPressure);
+	assert(memcmp(&priv, &base, sizeof(priv)) == 0);
```

Comment:
IMO this is wrong, it just happens to work because `minPressure` is also the min pressure threshold (1% of 2000 is 20). Or if that was the intention, it's confusing given the comment in line 1885. What I'd expect here is that a pressure below the *current minimum* is rebased down to a new minimum, but must now also be `>=` the `min_pressure_threshold`.

So same thing, but with `minPressure = 50` (or some other value > 20), `ds.pressure = 10` and then the value check should be that `pressure > ds.pressure` **and** `pressure < priv.minPressure`. Or, simpler, just have `const int min_pressure_threshold = 20;` in this test with a comment that it's based on the `wcmMaxZ = 2000` and then always compare against that. If we ever change the 0.01% in the driver, we can easily update this test too.

Reply:
The value was intentional. I wanted to make sure minPressure never falls below the min_pressure_threshold. The comment at line 1885 can be confusing. But the fact is the new minPressure is rebased, which should be the same as the min_pressure_threshold.

Reply:
>  I wanted to make sure minPressure never falls below the min_pressure_threshold

yes, that's a good check to make but it should be done as such - with the value chosen as it is you don't actually test that because you can't tell whether `minPressure` never changed or whether it got pushed down to `min_pressure_threshold`. Hence why `minPressure` must be higher than the threshold so you can verify that a) it gets pushed down as it should and b) that it doesn't end up as less than `min_pressure_threshold`.

Reply:
> `minPressure` must be higher than the threshold so you can verify that a) it gets pushed down as it should and b) that it doesn't end up as less than `min_pressure_threshold`

Do you suggest two unit test cases or just one? I'd think we need two. That's why I said your suggestion require another unit test case.

Reply:
yeah, better to split it up into two test cases